clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -263,6 +264,16 @@ public String deserialize(String topic, byte[] data) {
                     return super.deserialize(topic, data);
                 }
             }
+
+            @Override
+            public String deserialize(String topic, Headers headers, ByteBuffer data) {
+                if (i == recordIndex) {
+                    throw new SerializationException();
+                } else {
```

Comment:
you actually don't need the else there, because it will just throw.  It would be a bit cleaner if you do:
```
if (i == recordIndex) {
  throw new SerializationException();
}

i++;
return super.deserialize(topic, headers, data);
```

Reply:
I just kept the same style with the previous method `public String deserialize(String topic, byte[] data)`.